app/scripts/controllers/pane-articles-lists.js

```diff
@@ -0,0 +1,192 @@
+    'use strict';
+
+    /**
+    * Constructor function for the articles lists modal controller.
+    *
+    * @class ModalCtrl
+    */
+    function ModalCtrl($modalInstance, $sce, info, $rootScope) {
+        var self = this,
+            url;
+
+        switch(info.action) {
+            case 'edit':
+                url = [
+                    AES_SETTINGS.API.rootURI,
+                    '/admin/playlists/',
+                    info.articleId,
+                    '/',
+                    info.language,
+                    '/editor-view/', info.articlesListId
+                ].join('');
+
+                break;
+            case 'attach':
+                url = [
+                    AES_SETTINGS.API.rootURI,
+                    '/admin/playlists/',
+                    info.articleId,
+                    '/',
+                    info.language,
+                    '/editor-view'
+                ].join('');
+
+                break;
+            default:
+                url = '';
+        }
+
+        self.url = $sce.trustAsResourceUrl(url);
+
+        /**
+        * Closes the modal.
+        *
+        * @method close
+        */
+        self.close = function () {
+            $rootScope.$broadcast('close-articles-lists-modal');
+            $modalInstance.close();
+        };
+    }
+
+    /**
+    * AngularJS controller for the ArticlesLists pane.
+    *
+    * @class PaneArticlesListsCtrl
+    */
+    angular.module('authoringEnvironmentApp')
+        .controller('PaneArticlesListsCtrl', [
+        '$q',
+        '$modal',
+        '$scope',
+        'article',
+        'modalFactory',
+        'ArticlesList',
+        'toaster',
+        'TranslationService',
+        function (
+            $q,
+            $modal,
+            $scope,
+            articleService,
+            modalFactory,
+            ArticlesList,
+            toaster,
+            TranslationService) {
+            var self = this,
+                article = articleService.articleInstance;
+
+            // retrieve all articlesLists assigned to the article
+            self.assignedArticlesLists = ArticlesList.getAllByArticle(
+                article.articleId, article.language
+            );
+
+            $scope.$on('close-articles-lists-modal', function(event) {
+                ArticlesList.getAllByArticle(
+                    article.articleId, article.language
+                ).$promise.then(function (data) {
+                    // removes list from assignedArticlesLists
+                    // if the list doesn't exist in the response data
+                    angular.forEach(
+                        self.assignedArticlesLists,
+                        function (value, key) {
+                            if (!_.some(data, {id: value.id})) {
+                                _.remove(
+                                    self.assignedArticlesLists,
+                                    {id: value.id}
+                                );
```

Comment:
Just checking - is it really safe to delete the elements from the array while iterating over it? Does it, for example, work for two consecutive elements when both should be deleted? It is likely that the second one gets skipped over if its immediate predecessor gets deleted.

I would recommend creating a test case for this and modify the code if needed. One possible solution would be to iterate over the array _from the end_.